beam-postgres/Database/Beam/Postgres/PgSpecific.hs

```diff
@@ -410,6 +413,147 @@ isSubsetOf_ (QExpr needles) (QExpr haystack) =
 QExpr a ++. QExpr b =
   QExpr (pgBinOp "||" <$> a <*> b)
 
+-- | Postgres array_append(value) function.
+--
+-- Appends an element to the end of an array. Equivalent to the
+-- @anycompatiblearray || anycompatible@ operator.
+--
+-- Notes:
+-- - The array must be empty or one-dimensional (per Postgres rules for
+--   concatenating an element with an array).
+-- - If the array is NULL, the result is NULL. If the element is NULL,
+--   a NULL element is appended.
+arrayAppend_
+  :: QGenExpr ctxt Postgres s (V.Vector a)
+  -> QGenExpr ctxt Postgres s a
+  -> QGenExpr ctxt Postgres s (V.Vector a)
+arrayAppend_ (QExpr arr) (QExpr el) =
+  QExpr (PgExpressionSyntax . mappend (emit "array_append") . pgParens . mconcat <$> sequenceA
+    [ fromPgExpression <$> arr
+    , pure (emit ", ")
+    , fromPgExpression <$> el
+    ])
+
+-- | Postgres array_prepend(value) function.
+--
+-- Prepends an element to the beginning of an array. Equivalent to the
+-- @anycompatible || anycompatiblearray@ operator.
+--
+-- Notes:
+-- - The array must be empty or one-dimensional.
+-- - If the array is NULL, the result is NULL. If the element is NULL,
+--   a NULL element is prepended.
+arrayPrepend_
+  :: QGenExpr ctxt Postgres s a
+  -> QGenExpr ctxt Postgres s (V.Vector a)
+  -> QGenExpr ctxt Postgres s (V.Vector a)
+arrayPrepend_ (QExpr el) (QExpr arr) =
+  QExpr (PgExpressionSyntax . mappend (emit "array_prepend") . pgParens . mconcat <$> sequenceA
+    [ fromPgExpression <$> el
+    , pure (emit ", ")
+    , fromPgExpression <$> arr
+    ])
+
+-- | Postgres array_remove(value) function.
+--
+-- Removes all elements equal to the given value from the array.
+-- Comparisons use @IS NOT DISTINCT FROM@ semantics, so this can remove NULLs.
+--
+-- Notes:
+-- - The array must be one-dimensional.
+-- - Returns NULL only if the array is NULL; if the value is not present,
+--   the original array is returned unchanged.
+arrayRemove_
+  :: QGenExpr ctxt Postgres s (V.Vector a)
+  -> QGenExpr ctxt Postgres s a
+  -> QGenExpr ctxt Postgres s (V.Vector a)
+arrayRemove_ (QExpr arr) (QExpr el) =
+  QExpr (PgExpressionSyntax . mappend (emit "array_remove") . pgParens . mconcat <$> sequenceA
+    [ fromPgExpression <$> arr
+    , pure (emit ", ")
+    , fromPgExpression <$> el
+    ])
+
+-- | Postgres array_replace(array, from, to) function.
+--
+-- Replaces each element equal to the second argument with the third.
+--
+-- Notes:
+-- - Comparisons use IS NOT DISTINCT FROM semantics; can replace NULLs.
+-- - The array must be one-dimensional.
+--
+-- Example:
+--
+-- @
+-- select_ $ pure $ arrayReplace_ (val_ $ V.fromList [1::Int32,2,5,4]) (val_ 5) (val_ 3)
+-- -- => {1,2,3,4}
+-- @
+arrayReplace_
+  :: QGenExpr ctxt Postgres s (V.Vector a)
+  -> QGenExpr ctxt Postgres s a
```

Comment:
Could we add a docstring to show which element will be replaced, and which is the replacement? Something like

```haskell
arrayReplace_ 
    :: QGenExpr ctxt Postgres s (V.Vector a)
    -- | Element to be replaced
    -> QGenExpr ctxt Postgres s a
    -- | Replacement element
    -> QGenExpr ctxt Postgres s a
    -> QGenExpr ctxt Postgres s (V.Vector a)
```

I know this information is also present in the example, but having parameter docstrings is faster to read